### PR TITLE
app-containers/incus: fix cross compilation issue

### PR DESCRIPTION
**Pull request by tormath1**
Hi, in this PR we fix some cross-compilation issue in the Incus ebuild:
 * explicitly call the *src_unpack
 * set install path in function of the architecture

Let me know if I should squash the commits.

---
When cross-compiling to arm64, the Go environment variables were not properly set from the `go-env.eclass` because the `verify-sig` src_unpack method was overriding the src-unpack method from the `go-module.eclass`.

Reference: https://github.com/flatcar/scripts/pull/1655

### CODE REVIEW

**Review by juippis — COMMENTED** [review]
Let's only focus on 6.0.0-r1, older versions should've been dropped already from the tree. I can cherry-pick the 2 commits and squash them.

Can you show me a full build log from a successful build on arm64 system (that I guess is cross-compiled)?

File: app-containers/incus/incus-6.0.0-r1.ebuild [hunk 1]
```diff
@@ -147,7 +152,12 @@ src_test() {
 
 src_install() {
 	export GOPATH="${S}/_dist"
-	local bindir="_dist/bin"
+
+	if [[ "${GOARCH}" == "arm64" ]]; then
```
Comment (Flowdalic) [inline app-containers/incus/incus-6.0.0-r1.ebuild:156]: I wonder if this shoudn't be more generic:

```suggestion
	if is_crosscompile; then
```

(from toolchain.eclass)

File: app-containers/incus/incus-6.0.0-r1.ebuild [hunk 2]
```diff
@@ -176,11 +186,13 @@ src_install() {
...
-	"${bindir}"/incus completion bash > "${D}"/usr/share/bash-completion/completions/incus || die
-	"${bindir}"/incus completion fish > "${D}"/usr/share/fish/vendor_completions.d/incus.fish || die
-	"${bindir}"/incus completion zsh > "${D}"/usr/share/zsh/site-functions/_incus || die
+	if ! tc-is-cross-compiler; then
+		# Generate and install shell completion files.
+		mkdir -p "${D}"/usr/share/{bash-completion/completions/,fish/vendor_completions.d/,zsh/site-functions/} || die
+		"${bindir}"/incus completion bash > "${D}"/usr/share/bash-completion/completions/incus || die
+		"${bindir}"/incus completion fish > "${D}"/usr/share/fish/vendor_completions.d/incus.fish || die
+		"${bindir}"/incus completion zsh > "${D}"/usr/share/zsh/site-functions/_incus || die
+	fi
```
Comment (juippis) [inline app-containers/incus/incus-6.0.0-r1.ebuild:195]: You modify the `bindir` already, isn't this useless now? Whether or not crosscompiling it should resolve to by the former function.

Comment (juippis) [reply app-containers/incus/incus-6.0.0-r1.ebuild:195]: I'm gonna add an `ewarn` here about shell completion files not being installed. Rest looks good I guess, thanks!

**Review by juippis — APPROVED** [review]

**Comment by juippis** [comment]
https://bugs.gentoo.org/930496

I think I'm gonna try with @Flowdalic's suggestion about modifying the bindir only when tc-is-crosscompiler is detected. Please test it so it works once it's committed.

**Pull request closed without merging**